scripts/visualize-scripts/import-logs.py

```diff
@@ -8,21 +8,53 @@
 import requests
 import gnupg
 
+# when decrypting, ensure GPG_HOME_FOLDER exists
 GPG_HOME_FOLDER = "/root/.gnupg"
 UNENCRYPTED_LOGS_FOLDER = "/project/data/logs/http"
 
+# utils
 def strip_newlines(file):
     return list(line.rstrip('\n') for line in file)
 
-def preprocess_file(file_path, destination_folder, gpg_instance, passphrase):
+def batch_iterator(itor, batch_size):
+    line_index = 0
+    while True:
```

Comment:
Ken je Python's [range](https://docs.python.org/3/tutorial/controlflow.html#the-range-function)-functie (in combinatie met een for-loop) al? Kan je een alternatief voor de (iets moeilijker leesbare) infinite while-loop proberen uitwerken dmv `range`? Ik heb de oplossing hiermee nog niet in m'n hoofd. Laat gerust weten dat ik me vergis, wanneer een oplossing dmv `range` niet mogelijk lijkt.

Reply:
Kan je kijken of je er met [more_itertools.ichunked](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.ichunked ) komt? Die lijkt me "perfect for the job". (vergeet niet van `more_itertools` toe te voegen aan de [pip install](https://github.com/redpencilio/app-http-logger/blob/8faed2aa6bf1c83b8845a8cf5a61144f1a28d235/scripts/visualize-scripts/Dockerfile#L4))